src/store-api/src/storage/region.rs

```diff
@@ -110,10 +110,40 @@ pub struct FlushContext {
     /// If true, the flush will wait until the flush is done.
     /// Default: true
     pub wait: bool,
+    /// Flush reason.
+    pub reason: FlushReason,
 }
 
 impl Default for FlushContext {
     fn default() -> FlushContext {
-        FlushContext { wait: true }
+        FlushContext {
+            wait: true,
+            reason: FlushReason::Unknown,
+        }
+    }
+}
+
+/// Reason of flush operation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FlushReason {
+    /// Unknown reason.
+    Unknown,
+    /// Memtable is full.
+    MemtableFull,
+    /// Flush manually.
+    Manually,
+    /// Auto flush periodically.
+    Periodically,
+}
+
+impl FlushReason {
```

Comment:
Why not impl `fmt::Display`?

Reply:
Then we need to use format to get the string. But I expect `&'static str` so I could use it as metrics label.